tests/contrib/scrapy/test_pipeline_cerberus.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import
+import sys
+import os
+import pytest
+from scrapy import Item
+from scrapy.utils.test import get_crawler
+from spidermon.contrib.scrapy.pipelines import ItemValidationPipeline
+from tests.fixtures.items import TestItem, TreeItem
+from tests.fixtures.validators import (
+    cerberus_tree_schema,
+    cerberus_test_schema,
+    test_cerberus_schema_string,
+)
+
+STATS_ITEM_ERRORS = "spidermon/validation/items/errors"
+STATS_TYPES = "spidermon/validation/validators/{}/{}"
+SETTING_CERBERUS = "SPIDERMON_VALIDATION_CERBERUS"
+
+
+def assert_in_cerberus_stats(obj):
+    return "{}".format(STATS_TYPES.format(obj.__name__.lower(), "cerberus"))
+
+
+def the_stats(pipe):
+    return pipe.stats.stats.get_stats(pipe)
+
+
+@pytest.mark.parametrize(
+    "item,settings,cases",
+    [
+        pytest.param(
+            TestItem({"url": "example.com"}),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            [assert_in_cerberus_stats(Item)],
+            id="processing usual items without errors",
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: cerberus_test_schema}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, loads from a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [f"{os.getcwd()}/contrib/scrapy/schema.json"]},
+            [assert_in_cerberus_stats(Item), "{}".format(STATS_ITEM_ERRORS)],
+            id="validator is {} type, loads from path to schema".format(Item.__name__),
+        ),
+        pytest.param(
+            TestItem(),
+            {
+                SETTING_CERBERUS: {
+                    TestItem: "tests.fixtures.validators.cerberus_test_schema"
+                }
+            },
+            [assert_in_cerberus_stats(TestItem), "{}".format(STATS_ITEM_ERRORS)],
+            id="validator is {} type, loads from path to a python dict".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: {TestItem: [cerberus_test_schema]}},
+            [assert_in_cerberus_stats(TestItem), STATS_ITEM_ERRORS],
+            id="validator is {} type, validators are in a list repr".format(
+                TestItem.__name__
+            ),
+        ),
+        pytest.param(
+            TestItem(),
+            {SETTING_CERBERUS: [cerberus_test_schema]},
+            ["spidermon/validation/fields/errors/missing_required_field"],
+            id="Missing required fields",
+        ),
+    ],
+)
+def test_stats_in_pipeline(item, settings, cases):
+    crawler = get_crawler(settings_dict=settings)
+    pipe = ItemValidationPipeline.from_crawler(crawler)
+    pipe.process_item(item, None)
+    for case in cases:
+        casechecker = lambda x: x in the_stats(pipe)
```

Comment:
why not `assert x in the_stats(pipe)`

Reply:
Right regarding this, in some cases, we are testing 2 even 3 items to be present in stats. That's why I thought Lambda would be a great option. Changed.